site/content/en/docs/Tasks/namespaces_and_names.md

```diff
@@ -6,3 +6,245 @@ date: 2023-10-14
 ---
+
+The Namespace can be set for all Resources in a project by adding the [`namespace`] entry to the `kustomization.yaml` file. Consistent naming conventions can be applied to Resource Names in a project with the [`namePrefix`] and [`nameSuffix`] fields.
+
+## Working with Namespaces
+The Namespace for all namespaced Resources in a project can be set with the [`namespace`] field. This sets the Namespace for both generated Resources (e.g. ConfigMaps and Secrets) and non-generated Resources. This will override Namespace values that already exist.
+
+### Add Namespace
+The following example sets the Namespace of a Deployment and a generated ConfigMap.
+
+1. Create a Kustomization file.
+```yaml
+# kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: my-namespace
+
+configMapGenerator:
+- name: my-config
+  literals:
+  - FOO=BAR
+
+resources:
+- deploy.yaml
+```
+
+2. Create a Deployment manifest.
+```yaml
+# deploy.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: example
+```
+
+3. Add Namespace with `kustomize build`.
+```bash
+kustomize build .
+```
+
+The output shows that Namespace is added to the metadata field of the Deployment and generated ConfigMap.
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: example
+  namespace: my-namespace
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-config-m2mg5mb749
+  namespace: my-namespace
+data:
+  FOO: BAR
+```
+
+## Working with Names
+A prefix or suffix can be set for all Resources in a project with the `namePrefix` and `nameSuffix` fields. This sets a name prefix and suffix for both generated Resources (e.g. ConfigMaps and Secrets) and non-generated Resources.
+
+Resources such as Deployments and StatefulSets may reference other Resources such as ConfigMaps and Secrets in the Pod Spec. The name prefix and suffix will also propagate to Resource references in a project. Typical uses cases include Service reference from a StatefulSet, ConfigMap reference from a Pod Spec, and Secret reference from a Pod Spec.
+
+### Add Name Prefix
+The following example adds a prefix to the name of a Deployment and a generated ConfigMap.
+
+1. Create a Kustomization file.
+```yaml
+# kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namePrefix: foo-
+
+configMapGenerator:
+- name: my-config
+  literals:
+  - FOO=BAR
+
+resources:
+- deploy.yaml
+```
+
+2. Create a Deployment manifest.
+```yaml
+# deploy.yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: example
+```
+
+3. Add name prefix with `kustomize build`.
+```bash
+kustomize build .
+```
+
+The output shows that a prefix is added to the name of the Deployment and generated ConfigMap.
+```yaml
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: foo-my-config-m2mg5mb749
+data:
+  FOO: BAR
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: foo-example
+```
+
+### Add Name Suffix
```

Comment:
super nit: Currently, the `namePrefix` example includes a generated resource, but `nameSuffix` does not. Can we swap this?

I'm thinking it'd be cool if the examples increase in complexity. In other words, `namePrefix` applies to a pre-existing deployment, `nameSuffix` applies to a pre-existing deployment and generated ConfigMap, and the final example has all of the above in addition to the resource reference. As a reader, when I read the `nameSuffix` section, my reaction was `Why is there no generated resource here like all the other examples`?